packages/loader/container-loader/src/container.ts

```diff
@@ -1807,24 +1819,36 @@ export class Container
 		this.setLoaded();
 	}
 
-	private async rehydrateDetachedFromSnapshot(detachedContainerSnapshot: ISummaryTree) {
-		if (detachedContainerSnapshot.tree[hasBlobsSummaryTree] !== undefined) {
+	private async rehydrateDetachedFromSnapshot({
+		attached,
+		baseSnapshot,
+		snapshotBlobs,
+		hasAttachmentBlobs,
+	}: IPendingDetachedContainerState) {
+		if (hasAttachmentBlobs) {
 			assert(
 				!!this.detachedBlobStorage && this.detachedBlobStorage.size > 0,
-				0x250 /* "serialized container with attachment blobs must be rehydrated with detached blob storage" */,
+				"serialized container with attachment blobs must be rehydrated with detached blob storage",
 			);
-			// eslint-disable-next-line @typescript-eslint/no-dynamic-delete
-			delete detachedContainerSnapshot.tree[hasBlobsSummaryTree];
 		}
+		const snapshotTreeWithBlobContents: ISnapshotTreeWithBlobContents =
+			recombineSnapshotTreeAndSnapshotBlobs(baseSnapshot, snapshotBlobs);
+		this.storageAdapter.loadSnapshotFromSnapshotBlobs(snapshotBlobs);
+		return this.rehydrateDetachedFromSnapshotWithBlobs(snapshotTreeWithBlobContents);
```

Comment:
lets just combine this methods with the next one, i don't think we need them both